flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ExecutionGraphBuilder.java

```diff
@@ -179,6 +179,13 @@ public static ExecutionGraph buildGraph(
 		executionGraph.setScheduleMode(jobGraph.getScheduleMode());
 		executionGraph.setQueuedSchedulingAllowed(jobGraph.getAllowQueuedScheduling());
 
+		try {
+			executionGraph.setInputDependencyConstraint(
+                jobGraph.getSerializedExecutionConfig().deserializeValue(classLoader).getInputDependencyConstraint());
```

Comment:
I understand that this was the easiest way how to get the `InputDependencyConstraint` into the `ExecutionGraph` but I think it should be part of the `JobVertex`, because it is not a global setting but rather controls how each vertex is scheduled.

Reply:
Good suggestion. I've move `InputDependencyConstraint ` to JobVertex.